pipeline/matcher/scripts/getMatcherGraph.ts

```diff
@@ -0,0 +1,191 @@
+
+  return {
+    canonicalId: id,
+    pipelineDate: pipelineDate,
+  };
+}
+
+async function queryAllMatchingItems(client: DynamoDBDocument, query: QueryCommandInput): Promise<Array<Record<string, any>>> {
+  let allItems = [];
+
+  while (true) {
+    const result = await client.query(query);
+
+    if (!result.Items || result.Items.length == 0) {
+      break;
+    }
+
+    if (result.Items && result.Items.length > 0) {
+      allItems = [...allItems, ...result.Items]    
+    }
+
+    if (typeof result.LastEvaluatedKey === 'undefined') {
+      break;
+    }
+
+    query.ExclusiveStartKey = result.LastEvaluatedKey;
+  }
+
+  return allItems
+}
+
+async function getRelevantWorks(client: DynamoDBDocument, input: UserInput): Promise<SourceWork[]> {
+  const tableName = `catalogue-${input.pipelineDate}_works-graph`;
+
+  const output = await client.get({
+    TableName: tableName,
+    Key: { id: input.canonicalId }
+  });
+
+  if (typeof output.Item === 'undefined') {
+    throw new Error(`Could not find a matched work with ID ${input.canonicalId}`);
+  }
+
+  const subgraphId = output.Item!.subgraphId;
+
+  const query: QueryCommandInput = {
+    TableName: tableName,
+    IndexName: 'work-sets-index',
+    KeyConditionExpression: '#subgraphId = :subgraphId',
+    ExpressionAttributeNames: {'#subgraphId': 'subgraphId'},
+    ExpressionAttributeValues: {':subgraphId': subgraphId},
+  };
+
+  const worksInSubgraph: Array<Record<string, any>> = await queryAllMatchingItems(client, query);
+
+  return worksInSubgraph.map((item: Record<string, any>) => {
+    return {
+      canonicalId: item.id,
+      mergeCandidateIds: item.sourceWork.mergeCandidateIds,
+      suppressed: item.sourceWork.suppressed,
+      componentIds: item.componentIds,
+      sourceIdentifier: {
+        value: item.sourceWork.id.value,
+
+        // This is an artefact of a slightly weird format in DynamoDB,
+        // where the identifierType is recorded as e.g.
+        //
+        //    {"identifierType": {"METS": "METS"}, ...}
+        //
+        // We might simplify this at some point.
+        identifierType: Object.keys(item.sourceWork.id.identifierType)[0],
+      }
+    };
+  });
+}
+
+async function createGraph(works: SourceWork[]): Promise<RootCluster> {
+  const g = digraph('G',
+    
+    // This tells Graphviz to lay out the nodes left-to-right, because we tend
+    // to have monitors that are wider than tall, so it's easier to lay things
+    // out this way.
+    // See https://graphviz.org/docs/attrs/rankdir/
+    { rankdir: 'LR' },
+  );
+
+  // Create all the nodes
+  const nodes: Map<string, INode> = new Map();
+  await Promise.all(works.map(async (w: SourceWork) => {
+    let attributes = await getAttributes(w);
+
+    const newNode = g.createNode(w.canonicalId, attributes);
+
+    nodes.set(w.canonicalId, newNode);
+  }));
+
+  // Add all the edges
+  works.map((w: SourceWork) => {
```

Comment:
Since we're not using the return value, let's be clear that there are side effects
```suggestion
  works.forEach((w: SourceWork) => {
```